### ISSUE TITLE
README.md: Add clearer user instructions

### ISSUE THREAD

**Issue by li-boxuan** [coala/coala-vs-code#29]
Currently, the language server doesn't respond to ```textDocument/didChange``` requests. This may confuse users as users would expect to see feedback when they have edited their source code. Since this feature is not implemented yet (users would only see feedback after they saved the file), it would be better if we can inform users of that.

I know this is an expected feature for the [GSoC project](https://projects.coala.io/#/projects?project=coala_language_server&lang=en), but since coala plugin for VS Code is valid on the VS Code marketplace (without necessary instructions), it may be worthwhile to update README.md.

**Comment by gaocegege** [comment]
Yeah the documentation is outdated and we should maintain it better.
